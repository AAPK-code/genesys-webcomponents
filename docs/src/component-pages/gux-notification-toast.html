<html>
  <head>
    <title>gux-notification-toast</title>
  </head>
  <body>
    <script src="${CDN_URL}index.js"></script>
    <script>
      webcomponentsDocsMain(`
<gux-notification-toast
    toast-title='Holiday approved'
    icon='ic-calendar-generic'
    accent='neutral'
    message='Family vacation on 06/24/16 approved'>
</gux-notification-toast>
`);
    </script>
  </body>
</html>
