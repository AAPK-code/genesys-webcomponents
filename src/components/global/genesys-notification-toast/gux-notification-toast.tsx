import { Component, Event, EventEmitter, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-notification-toast.less',
  tag: 'gux-notification-toast'
})
export class GenesysNotificationToast {
  @Event()
  closeClick: EventEmitter;

  /**
   * The component accent (alert warning positive or neutral).
   */
  @Prop()
  accent: string = 'neutral';

  /**
   * The toast title.
   */
  @Prop()
  toastTitle: string;
  /**
   * The id of the title.
   */
  @Prop()
  idToast: string;
  /**
   * The icon name of the title.
   */
  @Prop()
  icon: string;
  /**
   * The icon uri of the title.
   */
  @Prop()
  iconUri: string;
  /**
   * The message of the toast.
   */
  @Prop()
  message: string;

  /**
   * Indicate the localisation strings
   */
  @Prop()
  i18n: { [key: string]: string } = {
    close: 'close'
  };

  onCloseButtonClickHandler(e) {
    this.closeClick.emit({ idToast: this.idToast, originalEvent: e });
  }

  /**
   * This function is to check accent
   */
  getAccent() {
    const accent = this.accent.toLowerCase();
    const allowed = ['warning', 'alert', 'neutral', 'positive'];
    return allowed.includes(accent) ? accent : 'neutral';
  }

  getIcon() {
    return this.icon ? (
      <i class={this.icon} />
    ) : this.iconUri ? (
      <img src={this.iconUri} />
    ) : (
      ''
    );
  }

  render() {
    return (
      <div class="gux-notification-toast">
        <div class={'icon-wrapper ' + this.getAccent()}>{this.getIcon()}</div>
        <div class="content-wrapper">
          <div class="title" title={this.toastTitle}>
            {this.toastTitle}
          </div>
          <div class="message" title={this.message}>
            {this.message}
          </div>
        </div>
        <div class="close-icon-wrapper">
          <button
            onClick={e => this.onCloseButtonClickHandler(e)}
            title={this.i18n.close}
          >
            <i class="genesys-icon-close" />
          </button>
        </div>
      </div>
    );
  }
}
