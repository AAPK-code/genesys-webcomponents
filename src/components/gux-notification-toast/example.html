<gux-notification-toast
    toast-title='Holiday approved'
    icon='ic-calendar-generic'
    accent='neutral'
    message='Family vacation on 06/24/16 approved'>
</gux-notification-toast>
