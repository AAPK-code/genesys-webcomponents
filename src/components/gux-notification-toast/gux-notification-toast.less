// Variables part

gux-notification-toast {
  > div {
    padding: 0 16px;
    display: flex;
    width: 336px;
    height: 78px;
    position: relative;
    border: 1px solid;
    .content-wrapper {
      flex: 1;
      overflow: hidden;
    }
    .icon-wrapper gux-icon,
    .icon-wrapper img {
      height: 32px;
      width: 32px;
      margin-right: 8px;
      margin-top: 16px;
    }
    .title {
      margin-top: 11px;
      .subhead-2();
      line-height: 30px;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    }
    .message {
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    }
    .close-icon-wrapper {
      position: absolute;
      top: 8px;
      right: 8px;
      button {
        cursor: pointer;
        border: none;
        background: none;
        padding: 0;
      }
      gux-icon {
        width: 12px;
        height: 12px;
      }
    }
  }
}

// Theming

// Dark
.gux-notification-toast-dark-theme {
  > div {
    background: @gux-off-white;
    color: @gux-grey-2;
    border-color: @gux-border;
    .message {
      color: @gux-pale-gray;
    }
    .close-icon-wrapper {
      button {
        color: @gux-pale-gray;
      }
    }
    .icon-wrapper.alert {
      color: @gux-alert;
    }
    .icon-wrapper.warning {
      color: @gux-warning;
    }
    .icon-wrapper.positive {
      color: @gux-positive;
    }
    .icon-wrapper.neutral {
      color: @gux-neutral;
    }
  }
}

.gux-dark-theme {
  gux-notification-toast {
    .gux-notification-toast-dark-theme();
  }
}
gux-notification-toast.gux-dark-theme {
  .gux-notification-toast-dark-theme();
}

// Light
.gux-notification-toast-light-theme {
  > div {
    background: @gux-panel;
    color: @gux-off-white;
    border-color: @gux-grey-2;
    .message {
      color: @gux-pale-gray-2;
    }
    .close-icon-wrapper {
      button {
        color: @gux-pale-gray-2;
      }
    }
    .icon-wrapper.alert {
      color: @gux-alert-2;
    }
    .icon-wrapper.warning {
      color: @gux-warning;
    }
    .icon-wrapper.positive {
      color: @gux-positive;
    }
    .icon-wrapper.neutral {
      color: @gux-neutral-2;
    }
  }
}

.gux-light-theme {
  gux-notification-toast {
    .gux-notification-toast-light-theme();
  }
}
gux-notification-toast.gux-light-theme {
  .gux-notification-toast-light-theme();
}

gux-notification-toast {
  .gux-notification-toast-light-theme();
}
