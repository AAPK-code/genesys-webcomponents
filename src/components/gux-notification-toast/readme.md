# gux-notification-toast

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description           | Type                                              | Default     |
| -------- | --------- | --------------------- | ------------------------------------------------- | ----------- |
| `accent` | `accent`  | The component accent. | `"alert" \| "neutral" \| "positive" \| "warning"` | `'neutral'` |


## Events

| Event        | Description | Type                |
| ------------ | ----------- | ------------------- |
| `guxdismiss` |             | `CustomEvent<void>` |


## Slots

| Slot        | Description                                      |
| ----------- | ------------------------------------------------ |
| `"icon"`    | Required slot for gux-icon                       |
| `"message"` | Required slot for the notification toast message |
| `"title"`   | Required slot for the notification toast title   |


## Dependencies

### Depends on

- [gux-icon](../gux-icon)

### Graph
```mermaid
graph TD;
  gux-notification-toast --> gux-icon
  style gux-notification-toast fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
