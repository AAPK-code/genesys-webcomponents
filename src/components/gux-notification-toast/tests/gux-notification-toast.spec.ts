import { newSpecPage } from '@stencil/core/testing';
import { GuxNotificationToast } from '../gux-notification-toast';

describe('gux-notification-toast', () => {
  let component: GuxNotificationToast;

  beforeEach(async () => {
    const page = await newSpecPage({
      components: [GuxNotificationToast],
      html: `<gux-notification-toast></gux-notification-toast>`,
      language: 'en'
    });

    component = page.rootInstance;
  });

  it('should build', async () => {
    expect(component).toBeInstanceOf(GuxNotificationToast);
  });

  describe('Class Logic', () => {
    describe('getAccent', () => {
      it('should return neutral if props/attr is not set correctly', () => {
        expect(component.getAccent()).toBe('neutral');
        component.accent = 'aaaaaa';
        expect(component.getAccent()).toBe('neutral');
      });

      it('should return accent in lowercase if props/attr is set correctly', () => {
        expect(component.getAccent()).toBe('neutral');
        component.accent = 'PoSitive';
        expect(component.getAccent()).toBe('positive');
      });
    });

    describe('getIcon', () => {
      it('should return icon and accent classname', () => {
        expect(component.getIcon()).toBe('');
        component.iconUri = 'test';
        // testing JSX
        expect((component.getIcon() as any).$tag$).toBe('img');
        expect((component.getIcon() as any).$attrs$.src).toBe('test');
        component.icon = 'test';
        expect((component.getIcon() as any).$tag$).toBe('gux-icon');
        expect((component.getIcon() as any).$attrs$.iconName).toBe('test');
      });
    });
  });
});
