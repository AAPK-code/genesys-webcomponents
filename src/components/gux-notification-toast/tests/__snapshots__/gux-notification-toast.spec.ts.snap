// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-notification-toast #render should render alert notification toast 1`] = `
<gux-notification-toast accent="alert">
  <!---->
  <div class="alert icon">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="content">
    <div class="title">
      <div slot="title">
        Title
      </div>
    </div>
    <div class="message">
      <div slot="message">
        This is the message
      </div>
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon iconname="ic-close" screenreadertext="Dismiss"></gux-icon>
  </button>
</gux-notification-toast>
`;

exports[`gux-notification-toast #render should render neutral notification toast 1`] = `
<gux-notification-toast accent="neutral">
  <!---->
  <div class="icon neutral">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="content">
    <div class="title">
      <div slot="title">
        Title
      </div>
    </div>
    <div class="message">
      <div slot="message">
        This is the message
      </div>
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon iconname="ic-close" screenreadertext="Dismiss"></gux-icon>
  </button>
</gux-notification-toast>
`;

exports[`gux-notification-toast #render should render positive notification toast 1`] = `
<gux-notification-toast accent="positive">
  <!---->
  <div class="icon positive">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="content">
    <div class="title">
      <div slot="title">
        Title
      </div>
    </div>
    <div class="message">
      <div slot="message">
        This is the message
      </div>
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon iconname="ic-close" screenreadertext="Dismiss"></gux-icon>
  </button>
</gux-notification-toast>
`;

exports[`gux-notification-toast #render should render warning notification toast 1`] = `
<gux-notification-toast accent="warning">
  <!---->
  <div class="icon warning">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="content">
    <div class="title">
      <div slot="title">
        Title
      </div>
    </div>
    <div class="message">
      <div slot="message">
        This is the message
      </div>
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon iconname="ic-close" screenreadertext="Dismiss"></gux-icon>
  </button>
</gux-notification-toast>
`;
